workspaces/util/src/chunkBySize.ts

```diff
@@ -2,7 +2,23 @@ export function* chunkBySize<T>(
   array: readonly T[],
   chunkSize: number,
 ): Generator<T[], void, void> {
+  if (
+    typeof chunkSize !== "number" ||
```

Comment:
We shouldn't need this check -- TypeScript already defends against this by declaring the type as number. If you want to handle untyped code, you can do that by simply inverting the rest of the if statement:

```ts
if (!Number.isInteger(chunkSize) || chunkSize < 1) {
```

Anything that's not a number is also not an integer.